contracts/modules/commons/submodules/auth/SequenceChainedSig.sol

```diff
@@ -10,36 +10,19 @@ import "../../ModuleSelfAuth.sol";
 import "../../ModuleStorage.sol";
 
 import "../../../../utils/LibBytesPointer.sol";
+import "../../../../utils/LibOptim.sol";
 
 
 abstract contract SequenceChainedSig is IModuleAuth, ModuleSelfAuth {
   using LibBytesPointer for bytes;
 
-  bytes32 public constant SET_IMAGEHASH_TYPEHASH = keccak256("SetImagehash(bytes32 imageHash,uint256 checkpoint)");
-  bytes32 internal constant LAST_AUTH_CHECKPOINT_KEY = keccak256("org.sequence.module.auth.submodule.prefixed.last.auth.checkpoint");
-
-  event SetLastCheckpoint(uint256 _checkpoint);
+  bytes32 public constant SET_IMAGEHASH_TYPEHASH = keccak256("SetImagehash(bytes32 imageHash)");
 
   error LowWeightChainedSignature(bytes _signature, uint256 threshold, uint256 _weight);
   error WrongChainedCheckpointOrder(uint256 _current, uint256 _prev);
-  error WrongFinalCheckpoint(uint256 _checkpoint, uint256 _current);
-
-  function _hashSetImagehashStruct(bytes32 _imageHash, uint256 _checkpoint) internal pure returns (bytes32) {
-    return keccak256(abi.encode(SET_IMAGEHASH_TYPEHASH, _imageHash, _checkpoint));
-  }
-
-  function setLastAuthCheckpoint(uint256 _checkpoint) external onlySelf {
-    ModuleStorage.writeBytes32(LAST_AUTH_CHECKPOINT_KEY, bytes32(_checkpoint));
-    emit SetLastCheckpoint(_checkpoint);
-  }
-
-  function getLastAuthCheckpoint() public view returns (uint256) {
-    return uint256(ModuleStorage.readBytes32(LAST_AUTH_CHECKPOINT_KEY));
-  }
 
-  struct SetImagehashStruct {
-    uint256 checkpoint;
-    bytes signature;
+  function _hashSetImagehashStruct(bytes32 _imageHash) internal pure returns (bytes32) {
+    return LibOptim.fkeccak256(SET_IMAGEHASH_TYPEHASH, _imageHash);
```

Comment:
I added the hash because I'm not sure of the side-effects of signing an `imageHash` directly, mostly with regards of a possible collision of an `imageHash` with some other message/digest.

Hashing the imageHash like this should enable us to detect on the client-side if a dapp is asking us to sign a lazy-octopus approval, because the message will start with `SET_IMAGEHASH_TYPEHASH`